src/comparsion.py

```diff
@@ -0,0 +1,159 @@
+        'atom': 'Type',
+        'file': 'File',
+        'line': 'Line',
+        'code': 'CLJ_Code'
+    }, inplace=True)
+
+    clojure_master["CLJ"] = '+'
+
+    clojure_master.drop(columns = ['offset'], inplace = True)
+
+    # Merge Clojure and CodeQL dataframes
+
+    for key, df in dataframes.items():
+        k = pd.merge(clojure_master[clojure_master["Type"] == key], df, on = ['File', 'Type', 'Line'], how = 'outer')
+        k = k[["Type", "File", "Line", "CLJ", "CQL", "CLJ_Code", "CQL_Column", "CQL_Code"]]
+        k = k.drop_duplicates(subset = ["Type", "File", "Line", "CLJ", "CQL"], keep = 'first')
+        k['CLJ'] = k['CLJ'].fillna('-')
+        k['CQL'] = k['CQL'].fillna('-')
+        dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+
+    # Load Coccinelle dataframes
+
+    coccinelle_dataframes = {}
+    for file_path in Path(coccinelle_directory).glob('*.csv'):
+        filename = Path(file_path).stem
+        dataframe_key = filename
+        p = pd.read_csv(file_path)
+        p.columns = ["Type", "File", "Line", "CNL_Column", "CNL_Code"]
+        p["File"] = p["File"].apply(lambda x: re.sub(coccinelle_system_path, '', x, count=1))
+        p["CNL"] = '+'
+        coccinelle_dataframes[dataframe_key] = p
+
+    for i in coccinelle_dataframes.values():
+        i['Type'] = i['Type'].map(combined_mapping)
+
+    # Merge Coccinelle DataFrames
+
+    for i in coccinelle_dataframes.values():
+        key = i.loc[0, "Type"]
+        toMerge = dataframes[key]
+        k = pd.merge(toMerge, i, on = ['File', 'Type', 'Line'], how = 'outer')
+        k['CLJ'] = k['CLJ'].fillna('-')
+        k['CQL'] = k['CQL'].fillna('-')
+        k['CNL'] = k['CNL'].fillna('-')
+        # Reordering columns
+        k = k[["Type", "File", "Line", "CLJ", "CQL", "CNL", "CLJ_Code", "CQL_Column", "CQL_Code", "CNL_Column", "CNL_Code"]]
+        k = k.drop_duplicates(subset = ["Type", "File", "Line", "CLJ", "CQL", "CNL"], keep = 'first')
+        # Float type For consistency with CQL_Column, which is forced as a float because of NaN values
+        k["CQL_Column"] = k["CQL_Column"].astype(float)
+        dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+
+    # Output merged CSVs to output_directory
+
+    for key, df in dataframes.items():
+        df = df.reindex(columns=["Type", "File", "Line", "CLJ", "CQL", "CNL", "CLJ_Code", "CQL_Column", "CQL_Code", "CNL_Column", "CNL_Code"])
+        df = pd.DataFrame(df)
+        if whichtwo == "CLJvCQL":
+            df = df.drop(['CNL', 'CNL_Column', 'CNL_Code'], axis = 1)
+        elif whichtwo == "CLJvCNL":
+            df = df.drop(['CQL', 'CQL_Column', 'CQL_Code'], axis = 1)
+            None
+        elif whichtwo == "CQJvCNL":
+            df = df.drop(['CLJ', 'CLJ_Code'], axis = 1)
+        else:
+            None
+        
+        df = df[df.apply(count_plus_specific_columns, axis=1, columns=which_two) >= 2]
```

Comment:
After dropping, we count the how many '+'s are present in the two or three columns (CLJ, CQL, and CNL). Regardless whether the user wants only two or keep all three, in the end we need at least two positives, so the sum >= 2.